Migrate to CP preinstall hooks

orta opened the pull request:
This reds locally, but works when I'm doing it with an app. Wanted this available for people to use as it's the fix for: #124

ashfurrow commented:
Cool, looks great! Merge whenevs.

marcelofabri commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
 ## 1.6.0
```
Version bump needed here

segiddins commented on lib/plugin.rb:
```diff
@@ -6,6 +6,11 @@ module CocoaPodsKeys
 
+    # Register for the pre-install hooks to setup & run Keys
+    Pod::HooksManager.register('cocoapods-keys', :pre_install) do
+      CocoaPodsKeys.setup
```
this should now call `setup` with the options, instead of pulling off of `Pod::Config.instance.podfile`

orta commented:
OK, addressed all feedback, tests are green ( and rubocop is happy ) - I've run this as a `:path` gem successfully against Eigen of this PR https://github.com/artsy/eigen/pull/1383
